compositions/aws-provider/elasticache-redis/rediscluster.yaml

```diff
@@ -0,0 +1,299 @@
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: xredisclusters.awsblueprints.io
+  labels:
+    awsblueprints.io/provider: aws
+    crossplane.io/xrd: xredisclusters.awsblueprints.io
+    service: elasticache
+    provider: aws
+spec:
+  compositeTypeRef:
+    apiVersion: awsblueprints.io/v1alpha1
+    kind: XRedisCluster
+  resources:
+    - base:
+        apiVersion: ec2.aws.crossplane.io/v1beta1
+        kind: SecurityGroup
+        metadata: {}
+        spec:
+          forProvider:
+            description: placeholder
+            egress:
+              - ipProtocol: tcp
+                fromPort: 6380
+                toPort: 6380
+                userIdGroupPairs:
+                - description: Allowing Cache port Outbound Access
+                  groupIdSelector:
+                    matchControllerRef: true
```

Comment:
This does not work unless https://github.com/crossplane/crossplane-runtime/commit/75dda897e5ff48951ab2f6c8a6836f6cb32eb844 is used. If it is used, the correct way to specify this is:
```
- description: Allowing Cache port Outbound Access
   groupIdSelector:
     matchControllerRef: true
     policy:
       resolution: Optional
       resolve: Always
```
This applies to ingress as well.